dspace-uclouvain/src/main/java/org/dspace/uclouvain/pdfAttestationGenerator/factory/PDFAttestationGeneratorFactoryImpl.java

```diff
@@ -0,0 +1,59 @@
+package org.dspace.uclouvain.pdfAttestationGenerator.factory;
+
+import java.sql.SQLException;
+import java.util.UUID;
+
+import org.dspace.content.Item;
+import org.dspace.content.service.ItemService;
+import org.dspace.core.Context;
+import org.dspace.services.factory.DSpaceServicesFactory;
+import org.dspace.uclouvain.core.utils.MetadataUtils;
+import org.dspace.uclouvain.pdfAttestationGenerator.configuration.PDFAttestationGeneratorConfiguration;
+import org.dspace.uclouvain.pdfAttestationGenerator.handlers.MasterThesisPdfAttestationGeneratorHandler;
+import org.dspace.uclouvain.pdfAttestationGenerator.handlers.PDFAttestationGeneratorHandler;
+import org.dspace.uclouvain.pdfAttestationGenerator.model.Handler;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class PDFAttestationGeneratorFactoryImpl implements PDFAttestationGeneratorFactory {
+    
+    @Autowired
+    ItemService itemService;
+
+    @Autowired
+    MasterThesisPdfAttestationGeneratorHandler masterThesisHandler;
+    
+    @Autowired
+    PDFAttestationGeneratorConfiguration pdfAttestationGeneratorConfiguration;
+
+    @Autowired
+    MetadataUtils metadataUtils;
+
+    /** 
+     * Returns an handler instance for a dspace item to handle PDF attestation generation.
+     * 
+     * @param uuid: The uuid of the target DSpace Item.
+     * @return An handler or null if not found.
```

Comment:
It could be better to raise an exception if no handler could be found (custom `HandlerNotFoundException`) and catch this exception in the parent code. This is a programmation principle : "Stay simple in method implementation, don't try to "help" parent method"